hudi-utilities/src/main/java/org/apache/hudi/utilities/transform/Transformer.java

```diff
@@ -45,4 +47,9 @@ public interface Transformer {
    */
   @PublicAPIMethod(maturity = ApiMaturityLevel.STABLE)
   Dataset<Row> apply(JavaSparkContext jsc, SparkSession sparkSession, Dataset<Row> rowDataset, TypedProperties properties);
+
+  @PublicAPIMethod(maturity = ApiMaturityLevel.EVOLVING)
+  default Option<Schema> transformedSchema(JavaSparkContext jsc, SparkSession sparkSession, Schema incomingSchema, TypedProperties properties) {
```

Comment:
I don't think it makes sense for this to return an Option. All rows will have a schema of some sorts so this option would never be empty in practice.

Reply:
This was done to avoid null handling. Please refer https://github.com/apache/hudi/pull/8574#discussion_r1176508297